src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/ValidationContext.cs

```diff
@@ -77,37 +86,57 @@ public class ValidationContext
                 throw new ArgumentNullException(nameof(sourceRepositories));
             }
 
-            if (logger == null)
-            {
-                throw new ArgumentNullException(nameof(logger));
-            }
-
             Package = package ?? throw new ArgumentNullException(nameof(package));
             Entries = entries?.ToList();
             DeletionAuditEntries = deletionAuditEntries.ToList();
             Client = client ?? throw new ArgumentNullException(nameof(client));
             CancellationToken = token;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _commonLogger = logger.AsCommon();
 
             _databasetimestampMetadataResource = sourceRepositories.V2.GetResource<IPackageTimestampMetadataResource>();
             _databasePackageRegistrationMetadataResource = sourceRepositories.V2.GetResource<IPackageRegistrationMetadataResource>();
             _v3PackageRegistrationMetadataResource = sourceRepositories.V3.GetResource<IPackageRegistrationMetadataResource>();
-
-            var commonLogger = logger.AsCommon();
+            _httpSourceResource = sourceRepositories.V3.GetResource<HttpSourceResource>();
 
             _databaseIndex = new Lazy<Task<PackageRegistrationIndexMetadata>>(
-                () => _databasePackageRegistrationMetadataResource.GetIndexAsync(Package, commonLogger, CancellationToken));
+                () => _databasePackageRegistrationMetadataResource.GetIndexAsync(Package, _commonLogger, CancellationToken));
             _v3Index = new Lazy<Task<PackageRegistrationIndexMetadata>>(
-                () => _v3PackageRegistrationMetadataResource.GetIndexAsync(Package, commonLogger, CancellationToken));
+                () => _v3PackageRegistrationMetadataResource.GetIndexAsync(Package, _commonLogger, CancellationToken));
 
             _databaseLeaf = new Lazy<Task<PackageRegistrationLeafMetadata>>(
-                () => _databasePackageRegistrationMetadataResource.GetLeafAsync(Package, commonLogger, CancellationToken));
+                () => _databasePackageRegistrationMetadataResource.GetLeafAsync(Package, _commonLogger, CancellationToken));
             _v3Leaf = new Lazy<Task<PackageRegistrationLeafMetadata>>(
-                () => _v3PackageRegistrationMetadataResource.GetLeafAsync(Package, commonLogger, CancellationToken));
+                () => _v3PackageRegistrationMetadataResource.GetLeafAsync(Package, _commonLogger, CancellationToken));
 
             _timestampMetadataDatabase = new Lazy<Task<PackageTimestampMetadata>>(
                 () => _databasetimestampMetadataResource.GetAsync(this));
         }
 
+        public async Task<IEnumerable<IPackageSearchMetadata>> GetSearchPageForIdAsync(Uri baseUri)
```

Comment:
>baseUri [](start = 91, length = 7)

It's a tiny bit unclear what `baseUri` is. What do you think of a name like `searchBaseUri`?